### ISSUE TITLE
[Enhancement] Export view as PDF

### ISSUE THREAD

**Issue by radeksvarz** [archimatetool/archi#626]
Use case: Bob wants to export 1 view into PDF from the model / .archimate file, which contains more views.

Current situation:  
 - Bob can export 1 view to the Image (right click - Export -> View as Image
 - Bob can export whole model into PDF (File -> Report -> Jasper ...)

Current workaround: 
Bob copies content of the particular view (CTRL A) into the new model.
Bob uses Jasper report to create the PDF file.
note: Bob has to set Jasper related attributes in the new model if needed. 

Desired situation:
Bob right clicks on the view (Export -> View as Jasper report), which would initiate the Jasper dialog with the scope of that particular view and model's elements, which are present on the view.

**Comment by Phillipus** [comment]
Did you try the Export View as Image and select the PDF option from the drop-down in the dialog?

**Comment by jbsarrodie** [comment]
Or Bob can put tags on views and run the Jasper report using the `#tag` selector...

See [Archi USer Guide](https://www.archimatetool.com/downloads/Archi%20User%20Guide.pdf) on page 90:

**Report:View:Tag**
>Use this property to tag a view (you can use this property multiple time). Tags can then be used to create a report containing only one specific tag. To do this add "#tagname" to the end of the report title (case insensitive). You can make a view visible for all tag using the special tag "all".

**Comment by Phillipus** [comment]
Shall we close this one? I think Bob has some good options, now.

**Comment by jbsarrodie** [comment]
> Shall we close this one? I think Bob has some good options, now.

One action (on my side) is be make it more clear that Jasper support this use case. I'll update user manual in HTTP for upcoming 4.7 (I've seen a new branch for that).

_Reopening until this is done_

**Comment by radeksvarz** [comment]
@Phillipus , @jbsarrodie Thanks for the answers. 

The added value of the PDF export is to report also the (non visible) content, such as descriptions, etc.

Therefore export -> view as image -> PDF file looses this added value. Though Bob appreciates that such convenient function is available ;)

I tested the tag option (thank you for mentioning that):
 - only the tagged view was exported, however for other content (relations, elements, etc.) the whole file's model was exported. Which makes the bloated PDF with data tables and one desired view.

Comparing to the mentioned workaround where only view related elements and relations were exported.

**Comment by jbsarrodie** [comment]
> * however for other content (relations, elements, etc.) the whole file's model was exported. Which makes the bloated PDF with data tables and one desired view.

Read the full page 90 for an answer ;-)

**Comment by Phillipus** [comment]
> I've seen a new branch for that

@jbsarrodie Yes, all help changes for 4.7 are in that branch. That means we can make multiple changes and correct our own errors. When done, I'll squash and cherry pick into master.

**Comment by radeksvarz** [comment]
> > * however for other content (relations, elements, etc.) the whole file's model was exported. Which makes the bloated PDF with data tables and one desired view.
> 
> Read the full page 90 for an answer ;-)

I might be blind - my apologies - I do not see the property which would state "limit the scope of reported elements to rendered views only"

Or do you mean "Report:View:Detailed" ?  That did not limit that.

I also tested: 
Report:Model:Hide:Relations: true @ model file level (resp. all properties set to hide, but Views, and ViewNumbering)
and
Report:View:Detailed: true @ view 

Which resulted in the limited report (view + related elements), but without related relations.

Ideal output scope is selected view + view related elements + view related relations.

**Comment by jbsarrodie** [comment]
> I might be blind - my apologies

No need to apology, this tends to proove that this part of the doc has to be rewritten.

You have to set all these properties to `true` at model level:
- Report:Model:Hide:Strategy
- Report:Model:Hide:Business
- Report:Model:Hide:Application
- Report:Model:Hide:Technology&Physical
- Report:Model:Hide:Motivation
- Report:Model:Hide:Implementation&Migration
- Report:Model:Hide:Other
- Report:Model:Hide:Relations

This will keep only the "Views" section which you can further filter using a tag. You can then set `Report:View:Detailed` to true to get a more complete list of elements.

> Which resulted in the limited report (view + related elements), **but without related relations.**

Yes, per view list of relations is not supported.

**Comment by radeksvarz** [comment]
> > Which resulted in the limited report (view + related elements), **but without related relations.**
> 
> Yes, per view list of relations is not supported.

So in fact the enhancement could be that this would become supported :)

I.e. view only property:
Report:View:Relations: true / false
